### PR TITLE
feat: Handle archived tricks on Trick Details page

### PR DESCRIPTION
**Pull request by JulianDietzel**
Addresses #324 (Does exactly what is asked in the issue)

### CODE REVIEW

File: src/components/stickable/trick/DeleteTrickDialog.vue [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      {{ t('triggerButton') }}
+    </DialogTrigger>
+    <DialogContent>
+      <DialogHeader>
+        <DialogTitle>{{ t('dialogue.title', { trickName: trickName }) }}</DialogTitle>
+        <DialogDescription>{{ t('dialogue.description') }}</DialogDescription>
+      </DialogHeader>
+      <DialogFooter>
+        <DialogClose class="font-medium text-sm p-2">{{ t('dialogue.cancelButton') }}</DialogClose>
+        <Button @click="deleteTrickIfPossible" variant="destructive" size="sm">
```
Comment (WaldemarLehner) [inline src/components/stickable/trick/DeleteTrickDialog.vue]: I dont see the benefit in the `IfPossible` suffix there

File: src/components/ui/dialog/DialogScrollContent.vue [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      <DialogContent
+        :class="
+          cn(
+            'relative z-50 grid w-full max-w-lg my-8 gap-4 border border-border bg-background p-6 shadow-lg duration-200 sm:rounded-lg md:w-full',
+            props.class
+          )
+        "
+        v-bind="forwarded"
+        @pointer-down-outside="
+          (event) => {
```
Comment (WaldemarLehner) [inline src/components/ui/dialog/DialogScrollContent.vue:42]: what does this do?

File: src/lib/database/daos/trick.ts [hunk 3]
```diff
@@ -393,21 +394,144 @@ export class Trick implements DbObject {
...
+      });
+
+      // Update other tricks variationOf
+      allTricks.forEach((trick) => {
+        if (!trick.variationOf) {
+          return;
+        }
+        if (!trick.variationOf.some((e) => primaryKeysMatch(e, originalPrimaryKey))) {
+          return;
+        }
```
Comment (WaldemarLehner) [inline src/lib/database/daos/trick.ts]:
```suggestion
        if ((!trick.variationOf || !trick.variationOf.some((e) => primaryKeysMatch(e, originalPrimaryKey))) {
          return;
        }
```

File: src/lib/database/daos/trick.ts [hunk 4]
```diff
@@ -393,21 +394,144 @@ export class Trick implements DbObject {
...
+
+      // Update all references in combos
+      const allCombos = await this.db.combos.toArray();
+      allCombos.forEach((combo) => {
+        if (!combo.tricks) {
+          return;
+        }
+        if (!combo.tricks.some((e) => primaryKeysMatch(e, originalPrimaryKey))) {
+          return;
+        }
```
Comment (WaldemarLehner) [inline src/lib/database/daos/trick.ts:531]:
```suggestion
        if (!combo.tricks || !combo.tricks.some((e) => primaryKeysMatch(e, originalPrimaryKey)) {
          return;
        }
```

**Review by WaldemarLehner — COMMENTED** [review]
some nits here and there. 

Yeah… in hindsight, that "Error handling" I did was a hot mess :'D

File: src/components/ui/dialog/DialogScrollContent.vue (hunk 2 above)
Comment (JulianDietzel) [reply src/components/ui/dialog/DialogScrollContent.vue:42]: No clue. I took this code from the shadcn-vue component library.

File: src/components/stickable/trick/DeleteTrickDialog.vue (hunk 1 above)
Comment (JulianDietzel) [reply src/components/stickable/trick/DeleteTrickDialog.vue]: Agreed. Removed the suffix. Changed the name to `delteTrick`.

File: src/lib/database/daos/trick.ts (hunk 3 above)
Comment (JulianDietzel) [reply src/lib/database/daos/trick.ts]: It makes for a fairly long line and might decrease readability by having one long if-statement which is harder to understand at a glance. But then also this functionally not very relevant (though necessary) section takes up less space this way. So I went with your suggestion and condensed the two ifs down to one. Same goes for your next comment.

**Comment by JulianDietzel** [comment]
Thanks a lot for the review @WaldemarLehner. I agreed with all your suggestions and implemented them. I found one small bug which I now need to fix and then I'm going to merge.

**Pull request merged**